m3/src/test/java/com/uber/m3/tally/m3/MockM3Service.java

```diff
@@ -43,23 +42,29 @@ public MockM3Service(CountDownLatch metricsCountLatch) {
 
     public List<MetricBatch> getBatches() {
         lock.readLock().lock();
-
         try {
             return batches;
         } finally {
             lock.readLock().unlock();
         }
     }
+    public List<Metric> getMetrics() {
+        lock.readLock().lock();
+        try {
+            return metrics;
```

Comment:
instead of reference to a guarded variable, should you return a copy here?

Reply:
This is just to silence the linter -- this is a test-code, method is only method to be called at the end of the test

Reply:
OK. Please add a `WARNING` in the method doc letting people know it's broken if used in any other way.

Reply:
It's not broken, this is a test method within the text package -- i can hardly imagine it's being used in any other way

Reply:
Simple failure mode: if users access the data during test while they're still emitting/changing metrics, it's racy.

Reply:
Fair enough, will update